spec/factories/ext_management_system.rb

```diff
@@ -0,0 +1,34 @@
+FactoryGirl.define do
+  factory :ems_nuage_with_vcr_authentication, :parent => :ems_nuage_network do
+    zone do
+      _guid, _server, zone = EvmSpecHelper.create_guid_miq_server_zone
+      zone
+    end
+
+    after(:build) do |ems|
+      ems.hostname = Rails.application.secrets.nuage_network.try(:[], 'host') || 'nuagenetworkhost'
+    end
+
+    after(:create) do |ems|
+      userid   = Rails.application.secrets.nuage_network.try(:[], 'userid') || 'NUAGE_USER_ID'
+      password = Rails.application.secrets.nuage_network.try(:[], 'password') || 'NUAGE_PASSWORD'
+
+      # Ensure that VCR will obfuscate the basic auth
+      VCR.configure do |c|
```

Comment:
With this factory girl I followed the example by the Amazon provider ([here](https://github.com/ManageIQ/manageiq-providers-amazon/blob/master/spec/factories/ext_management_system.rb#L9)). Since the actual Nuage backend and it's credentials should not be shared publicly, I decided to use a similar approach to the one in VMware cloud manager ([here](https://github.com/ManageIQ/manageiq-providers-vmware/blob/master/spec/models/manageiq/providers/vmware/cloud_manager/refresher_spec.rb#L20-L28)).

The problem that I have is that this mixes two aspects (factory model and VCR configuration) so I wonder if it would be better to

 * move just the VCR config into the [corresponding spec](https://github.com/ManageIQ/manageiq-providers-nuage/pull/6/files#diff-3fadeabfdfdb1f0e8c5b79d3188ac70aR11) file into it's own `before(:each)`? This would again read the hostname, userid and password from `Rails.application.secrets`
 * move everything into the [corresponding spec](https://github.com/ManageIQ/manageiq-providers-nuage/pull/6/files#diff-3fadeabfdfdb1f0e8c5b79d3188ac70aR11) (as VMware does)